### PR TITLE
Writing command line friendly applicaitons

### PR DESCRIPTION
**Pull request by tebeka**

### CODE REVIEW

File: content/advent-2019/cmdline.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+or has it?
+
+The [Lindy effect](https://en.wikipedia.org/wiki/Lindy_effect) is a concept
+that the future life expectancy of some non-perishable things like a technology
+or an idea is proportional to their current age. TL;DR - old technologies are
+here to stay.
+
+If you don't believe me, see:
+
+- [oh-my-zsh](https://github.com/ohmyzsh/ohmyzsh) having close to 100,000 starts on GitHub
```
Comment (philpearl) [inline content/advent-2019/cmdline.md]:
```suggestion
- [oh-my-zsh](https://github.com/ohmyzsh/ohmyzsh) having close to 100,000 stars on GitHub
```

File: content/advent-2019/cmdline.md [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+dir: false
+modified: 2019-11-27 11:49:03.280857863 +0200 IST
+$ ./finfo -json finfo.go
+{"dir":false,"mode":"-rw-r--r--","modified":"2019-11-27T11:49:03.280857863+02:00","size":783}
+```
+
+## Progress
+
+Some operations can take long time, one way to make them faster is not by
+optimizing the code but by showing a spinner/progress bar. Don't beleive me,
```
Comment (philpearl) [inline content/advent-2019/cmdline.md]:
```suggestion
optimising the code but by showing a spinner/progress bar. Don't believe me,
```

File: content/advent-2019/cmdline.md [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+here's an excerpt from a [Nielsen
```
Comment (philpearl) [inline content/advent-2019/cmdline.md]:
```suggestion
here's an excerpt from [Nielsen
```

File: content/advent-2019/cmdline.md [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+
+}
+```
+
+Run it and you'll see a small spinner going.
+
+
+### Progress Bar
+
+For progress bar, you'll probably need an external package such as
```
Comment (philpearl) [inline content/advent-2019/cmdline.md]:
```suggestion
For a progress bar, you'll probably need an external package such as
```

**Comment by tebeka** [comment]
Thanks @philpearl

File: content/advent-2019/cmdline.md [hunk 5]
```diff
@@ -0,0 +1,325 @@
++++
+title = "Writing Friendly Command Line Applications"
+date = "2019-12-08T00:00:00+00:00"
```
Comment (bketelsen) [inline content/advent-2019/cmdline.md]: Please change the date to 2019-12-02 for publishing tomorrow

Comment (tebeka) [reply content/advent-2019/cmdline.md]: Done

**Comment by bketelsen** [comment]
Thank you for the contribution to the advent series!!

**Comment by tebeka** [comment]
Thanks!

**Pull request merged**